src/test/java/com/vinylteam/vinyl/web/servlets/ConfirmationServletTest.java

```diff
@@ -0,0 +1,173 @@
+        verify(mockedConfirmationService).findByToken(null);
+        inOrderResponse.verify(mockedResponse).setContentType("text/html;charset=utf-8");
+        inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_NOT_FOUND);
+        inOrderResponse.verify(mockedResponse).getWriter();
+    }
+
+    @Test
+    void doPostSuccessfulLogIn() throws IOException {
+        //prepare
+        when(mockedRequest.getParameter("token")).thenReturn(confirmationToken.getToken().toString());
+        when(mockedRequest.getParameter("email")).thenReturn(userByLinkToken.getEmail());
+        when(mockedRequest.getParameter("password")).thenReturn("rightPassword");
+        when(mockedUserService.signInCheck(userByLinkToken.getEmail(), "rightPassword", confirmationToken.getToken())).thenReturn(Optional.of(userByLinkToken));
+        when(mockedRequest.getSession(true)).thenReturn(mockedHttpSession);
+        //when
+        confirmationServlet.doPost(mockedRequest, mockedResponse);
+        //then
+        inOrderRequest.verify(mockedRequest).getParameter("token");
+        inOrderResponse.verify(mockedResponse).setContentType("text/html;charset=utf-8");
+        inOrderRequest.verify(mockedRequest).getParameter("email");
+        inOrderRequest.verify(mockedRequest).getParameter("password");
+        inOrderUserService.verify(mockedUserService).signInCheck(eq(userByLinkToken.getEmail()), eq("rightPassword"), eq(confirmationToken.getToken()));
+        inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_OK);
+        inOrderResponse.verify(mockedResponse).sendRedirect("/");
+    }
+
+    @Test
+    void doPostFailedLogIn() throws IOException {
+        //prepare
+        when(mockedRequest.getParameter("token")).thenReturn(confirmationToken.getToken().toString());
+        when(mockedRequest.getParameter("email")).thenReturn(userByLinkToken.getEmail());
+        when(mockedRequest.getParameter("password")).thenReturn("wrongPassword");
+        when(mockedUserService.signInCheck(userByLinkToken.getEmail(), "wrongPassword", confirmationToken.getToken())).thenReturn(Optional.empty());
+        when(mockedResponse.getWriter()).thenReturn(printWriter);
+        //when
+        confirmationServlet.doPost(mockedRequest, mockedResponse);
+        //then
+        inOrderRequest.verify(mockedRequest).getParameter("token");
+        inOrderResponse.verify(mockedResponse).setContentType("text/html;charset=utf-8");
+        inOrderRequest.verify(mockedRequest).getParameter("email");
+        inOrderRequest.verify(mockedRequest).getParameter("password");
+        inOrderUserService.verify(mockedUserService).signInCheck(eq(userByLinkToken.getEmail()), eq("wrongPassword"), eq(confirmationToken.getToken()));
+        inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        inOrderResponse.verify(mockedResponse).getWriter();
+    }
+
+    @Test
+    void doPostWrongEmail() throws IOException {
```

Comment:
Unfortunately, I couldn't find out: what happens when user logs in successfully, but it's not their token?

Reply:
user will be not loged in in this case, but thank you for your question - I will check and add test

Reply:
![image](https://user-images.githubusercontent.com/61110701/121483076-7353fb00-c9d6-11eb-9ac3-5bc6ed584f7d.png)

Reply:
DefaultUserServiceTest.signInWithTokenNotCorrectUser - test